### PR TITLE
Add setShowcaseEntitlements API

### PR DESCRIPTION
**Pull request by mborof**

### CODE REVIEW

File: src/api/subscriptions.js [hunk 1]
```diff
@@ -289,8 +289,38 @@ export class Subscriptions {
...
+/**
+* PublisherEntitlement
+* Properties:
+* - isUserRegistered: Is the user registered currently?
+* - entitlements: An array of relevant publisher entitlement events.
+*  @typedef {{
+  *    isUserRegister:  !boolean,
+  *    entitlements: !PublisherEntitlementEvents,
+  * }}
+  */
```
Comment (umang-google) [inline src/api/subscriptions.js:321]: indent

File: src/api/subscriptions.js [hunk 2]
```diff
@@ -289,8 +289,38 @@ export class Subscriptions {
...
+};
+
+/**
+* PublisherEntitlement
+* Properties:
+* - isUserRegistered: Is the user registered currently?
+* - entitlements: An array of relevant publisher entitlement events.
+*  @typedef {{
+  *    isUserRegister:  !boolean,
+  *    entitlements: !PublisherEntitlementEvents,
```
Comment (jpettitt) [inline src/api/subscriptions.js:319]: Not an array - can only be one at any given time as state are non overlapping.

File: src/api/subscriptions.js [hunk 3]
```diff
@@ -289,8 +289,38 @@ export class Subscriptions {
...
+  setShowcaseEntitlements(entitlements) {}  
 }
 
+/** @enum {string} */
+export const PublisherEntitlementEvents = {
+  EVENT_SHOWCASE_UNLOCKED_BY_SUBSCRIPTION: 'EVENT_SHOWCASE_UNLOCKED_BY_SUBSCRIPTION', // Publisher managed subscriptions only
+  EVENT_SHOWCASE_UNLOCKED_BY_METER: 'EVENT_SHOWCASE_UNLOCKED_BY_METER',               // Publisher managed meters only
+  EVENT_SHOWCASE_UNLOCKED_FREE_PAGE: 'EVENT_SHOWCASE_UNLOCKED_FREE_PAGE',             // Must be reported by publisher
+  
+  EVENT_SHOWCASE_NO_ENTITLEMENTS_REGWALL: 'EVENT_SHOWCASE_NO_ENTITLEMENTS_REGWALL',  // Generated by swg-gaa or publisher
```
Comment (jpettitt) [inline src/api/subscriptions.js:308]: This comment and the next one are confusing - just have the pub report them, worst case it's a duplicate which we can detect with the nonce,

File: src/api/subscriptions.js [hunk 4]
```diff
@@ -289,8 +289,39 @@ export class Subscriptions {
...
+
+/**
+* PublisherEntitlement
+*   In order to participate in News Showcase, publishers must report information about their entitlements.
+* Properties:
+* - isUserRegistered: Is the user registered currently?
+* - entitlement: Publisher entitlement event type.
+*  @typedef {{
+*    isUserRegister:  !boolean,
+*    entitlement: !PublisherEntitlementEvents,
```
Comment (jpettitt) [inline src/api/subscriptions.js:320]: PublisherEntitlementEvents -> PublisherEntitlementEvent

**Review by jpettitt — APPROVED** [review]

File: src/api/subscriptions.js (hunk 2 above)
Comment (jpettitt) [inline src/api/subscriptions.js]: entitlements -> entitlement

File: src/runtime/runtime.js [hunk 5]
```diff
@@ -1084,6 +1094,7 @@ function createPublicRuntime(runtime) {
     getPropensityModule: runtime.getPropensityModule.bind(runtime),
     getLogger: runtime.getLogger.bind(runtime),
     getEventManager: runtime.getEventManager.bind(runtime),
+    setShowcaseEntitlements: runtime.setShowcaseEntitlements.bind(runtime),
```
Comment (jpettitt) [inline src/runtime/runtime.js]: setShowcaseEntitlements setShowcaseEntitlement

**Comment by jpettitt** [comment]
I've LGTM'd but please fix all the plurals before merging.

**Pull request merged**